typescript/vrotsc/src/compiler/transformer/fileTransformers/workflow/decorators/switchItemDecoratorStrategy.ts

```diff
@@ -0,0 +1,410 @@
+		const lines = ["// Generated by the system, cannot be edited"];
+		
+		if (switchBag.cases && switchBag.cases.length > 0) {
+			switchBag.cases.forEach((caseItem, index) => {
+				const targetItem = super.findTargetItem(caseItem.target, pos, itemInfo);
+				const condition = this.formatConditionForScript(caseItem);
+				
+				if (index === 0) {
+					lines.push(`if (${condition}) {`);
+				} else {
+					lines.push(`} else if (${condition}) {`);
+				}
+				lines.push(`  return "${targetItem}";`);
+			});
+		}
+
+		// Add default case
+		if (switchBag.defaultTarget) {
+			const defaultTargetItem = super.findTargetItem(switchBag.defaultTarget, pos, itemInfo);
+			lines.push(`} else if (true) {`);
+			lines.push(`  return "${defaultTargetItem}";`);
+		}
+
+		if (lines.length > 1) {
+			lines.push("}");
+		}
+
+		return lines.join("\n");
+	}
+
+	/**
+	 * Formats a case condition for use in the generated JavaScript script.
+	 * 
+	 * Handles different condition types (string, number, boolean) and generates
+	 * appropriate comparison expressions. String conditions are wrapped in quotes,
+	 * while numbers and booleans are used directly.
+	 * 
+	 * @private
+	 * @param {any} caseItem - The case item containing variable and condition properties
+	 * @param {string} caseItem.variable - The variable name to compare (defaults to "value")
+	 * @param {any} caseItem.condition - The condition value to compare against
+	 * @returns {string} The formatted condition expression for the script
+	 */
+	private formatConditionForScript(caseItem: any): string {
+		const variable = caseItem.variable || "value";
+		const condition = caseItem.condition;
+		
+		if (typeof condition === "string") {
+			return `${variable} === "${condition}"`;
+		} else if (typeof condition === "number") {
+			return `${variable} === ${condition}`;
+		} else if (typeof condition === "boolean") {
+			return condition.toString();
+		}
+		
+		return `${variable} === ${JSON.stringify(condition)}`;
+	}
+
+	/**
+	 * Adds XML condition elements to the workflow item definition.
+	 * 
+	 * Creates `<condition>` XML elements for each case in the switch, including:
+	 * - Variable name and type information
+	 * - Comparator type (mapped from user-friendly strings to numeric codes)
+	 * - Target label for workflow navigation
+	 * - Condition value
+	 * 
+	 * @private
+	 * @param {StringBuilderClass} stringBuilder - The string builder to append XML to
+	 * @param {CanvasItemPolymorphicBagForSwitch} switchBag - The switch configuration bag
+	 * @param {number} pos - The position of the item in the workflow
+	 * @param {WorkflowItemDescriptor} itemInfo - The workflow item descriptor
+	 */
+	private addConditionElements(stringBuilder: StringBuilderClass, switchBag: CanvasItemPolymorphicBagForSwitch, pos: number, itemInfo: WorkflowItemDescriptor): void {
+		if (switchBag.cases) {
+			switchBag.cases.forEach(caseItem => {
```

Comment:
Same here, i think the if-case can be removed and we can use:

```suggestion
			(switchBag.cases || []).forEach(caseItem => {
```